plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java

```diff
@@ -857,6 +859,8 @@ protected HiveConfig getHiveConfig()
     {
         return new HiveConfig()
                 .setMaxOpenSortFiles(10)
+                // Use separate staging directory for each test class to prevent intermittent failures coming from test parallelism
+                .setTemporaryStagingDirectoryPath("/tmp/trino-${USER}-" + UPPER_CAMEL.converterTo(LOWER_HYPHEN).convert(this.getClass().getSimpleName()))
```

Comment:
Can this just be randomized, e.g. UUID based?

Also, for tmp local resources, we should also

```
deleteRecursively(..., ALLOW_INSECURE)
```